### PR TITLE
Replace cucumber tests with rspec system specs

### PR DESCRIPTION
**Pull request by t27duck**
## The Problem

The existing feature tests pretty much required the cucumber and related gems to be locked at their older versions due to changes in the syntax used for manipulating the browser.

That said, current-day Rails now has capybara integration for browser (system) tests baked in. `rspec-rails` can also work with it.

## The Solution

I ripped out all traces of cucumber and gherkin from the app and replaced the tests with Rails system tests through rspec.

I also "reinitialized" the rspec configuration so that it matches more with the default config from `rspec-rails`.

Finally, I cleaned up some of the existing specs and added a few request specs for some of the more common endpoints that are hit in the app.

I am not sure about how the old cucumber tests functioned, but at least with this PR, this adds a requirement of Google Chrome to be on the system that runs the tests. Chrome is the default used by capybara now. That said, for CI and WSL2, it needs to run headless, so I had to add a custom capybara driver which sent the correct flags to the Chrome binary to run in headless mode.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -48,16 +48,17 @@ group :development, :test do
 
   gem 'pry-rails'
   gem 'rb-readline'
+
+  gem 'rspec-rails', '~> 4.0.1'
```
Comment (t27duck) [inline Gemfile:52]: This goes in the dev and test groups as when you run "rake spec" you default load to the development environment. Rspec handles switching to the test environment when it starts up and loads the rest of the gems.

File: Gemfile [hunk 2]
```diff
@@ -48,16 +48,17 @@ group :development, :test do
...
 end
 
 group :test do
-  gem 'database_cleaner'
```
Comment (t27duck) [inline Gemfile:54]: "But how is the database cleaned up between tests?"

Rails handles that for us in its test suite via db checkpoints and transactions. It's handling of capybara also allows it to have a fresh db per test while cucumber needed DB cleaner due to the web server not talking to the actual test runner.

File: Gemfile [hunk 3]
```diff
@@ -48,16 +48,17 @@ group :development, :test do
...
-  gem 'rspec-rails'
-  gem 'gherkin'
+  # Adds support for Capybara system testing and selenium driver
```
Comment (t27duck) [inline Gemfile:56]: These lines are a copy/paste from the default Rails Gemfile

File: Rakefile [hunk 4]
```diff
@@ -6,6 +6,4 @@ require 'rake'
 
 IndyhackersRails::Application.load_tasks
 
-Rake::Task['test'].clear
```
Comment (t27duck) [inline Rakefile:9]: I originally removed these lines in my original attempt to figure out why specs were running twice. If someone feels strong enough about having `rake test` run specs, I can see about putting these back.

File: app/views/site/about.html.erb [hunk 5]
```diff
@@ -103,7 +103,7 @@ board_members = [
     <div class="card blank"></div>
     <% else %>
     <div class="card">
-      <%= image_tag "board/#{member[:picture]}", alt: member[:name], class: 'card-img-top' %>
+      <%= image_tag "board/#{member[:picture]}", skip_pipeline: true, alt: member[:name], class: 'card-img-top' %>
```
Comment (t27duck) [inline app/views/site/about.html.erb:106]: CI started throwing deprecation warnings about this that I couldn't reproduce locally. 🤷‍♂️

File: spec/requests/admin/jobs_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    describe "when not authenticated" do
+      it "asks the user to log in" do
+        patch admin_job_path(job), params: { job: job_params }
+
+        expect(response).to redirect_to(new_admin_session_path)
+      end
+    end
+  end
+
+  describe "#destroy" do
```
Comment (t27duck) [inline spec/requests/admin/jobs_controller_spec.rb:145]: The route works. The route is exposed (granted, behind a login), but there's nothing in the app that I can see that actually lets an admin delete a job. But, I spec-ed it out anyway. Later, I hope to go through and try to trim some of the unused routes down to get a better idea of what is actually available.

File: spec/requests/site_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,43 @@
+require "rails_helper"
+
+RSpec.describe SiteController do
```
Comment (t27duck) [inline spec/requests/site_controller_spec.rb:3]: I didn't cover everything in here, but I did focus on the urls that are most likely semi-permanent.

File: spec/support/capybara.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+Capybara.register_driver :headless_chrome do |app|
```
Comment (t27duck) [inline spec/support/capybara.rb:1]: Capybara has a built-in headless chrome driver, but it doesn't use enough flags to satisfy Travis-CI or WSL2 as they don't have `$DISPLAY`. This setup really makes it headless.

File: .travis.yml [hunk 9]
```diff
@@ -2,12 +2,14 @@ language: ruby
...
   - postgresql
 before_script:
   - psql -c 'create database indyhackers_test;' -U postgres
   - cp config/database.yml.travis config/database.yml
 
 script:
-  - RAILS_ENV=test bundle exec rake --trace db:migrate spec
-  - RAILS_ENV=test bundle exec rake --trace features
+  - bundle exec rake --trace db:setup
+  - bundle exec rake --trace spec
```
Comment (t27duck) [inline .travis.yml:15]: I could probably have this be one command, but I figured it would allow us to have other "steps" we can run.

**Comment by t27duck** [comment]
@armilam - Don't think I can think of anything else I'd want to do with this branch. So, whenever you get bored, enjoy this little mess I made.

File: spec/requests/admin/jobs_controller_spec.rb (hunk 6 above)
Comment (mileszs) [reply spec/requests/admin/jobs_controller_spec.rb:145]: There is a link on the admin index that allows one to delete a job.

Comment (t27duck) [reply spec/requests/admin/jobs_controller_spec.rb:145]: Good to know. Must have missed it.

**Review by mileszs — APPROVED** [review]

**Pull request merged**